src/api/java/com/enderio/api/integration/Integration.java

```diff
@@ -0,0 +1,23 @@
+package com.enderio.api.integration;
+
+import net.minecraftforge.eventbus.api.IEventBus;
+import net.minecraftforge.forge.event.lifecycle.GatherDataEvent;
+
+public abstract class Integration implements IntegrationMethods{
+
+    private String modid;
+
+    public String getModid() {
+        return modid;
+    }
+
+    public void setModid(String modid) {
+        if (this.modid != null)
+            throw new IllegalCallerException("You are not allowed to set the modid of an integration");
```

Comment:
I'm a little confused why we are handling it like this. Can you walk me through this either here or in discord?